components/style/properties/longhand/font.mako.rs

```diff
@@ -426,53 +522,62 @@ ${helpers.single_keyword("font-variant-caps",
                 }
-                LengthOrPercentage::Length(NoCalcLength::ServoCharacterWidth(value)) => {
+                SpecifiedValue::Length(LengthOrPercentage::Length(
+                        NoCalcLength::ServoCharacterWidth(value))) => {
                     value.to_computed_value(context.inherited_style().get_font().clone_font_size())
                 }
-                LengthOrPercentage::Length(ref l) => {
+                SpecifiedValue::Length(LengthOrPercentage::Length(ref l)) => {
                     l.to_computed_value(context)
                 }
-                LengthOrPercentage::Percentage(Percentage(value)) => {
+                SpecifiedValue::Length(LengthOrPercentage::Percentage(Percentage(value))) => {
                     context.inherited_style().get_font().clone_font_size().scale_by(value)
                 }
-                LengthOrPercentage::Calc(ref calc) => {
+                SpecifiedValue::Length(LengthOrPercentage::Calc(ref calc)) => {
                     let calc = calc.to_computed_value(context);
                     calc.length() + context.inherited_style().get_font().clone_font_size()
                                            .scale_by(calc.percentage())
                 }
+                SpecifiedValue::Keyword(ref key) => {
+                    key.to_computed_value(context)
+                }
             }
         }
 
         #[inline]
         fn from_computed_value(computed: &computed_value::T) -> Self {
-                SpecifiedValue(LengthOrPercentage::Length(
+                SpecifiedValue::Length(LengthOrPercentage::Length(
                         ToComputedValue::from_computed_value(computed)
                 ))
         }
     }
     /// <length> | <percentage> | <absolute-size> | <relative-size>
-    pub fn parse(_context: &ParserContext, input: &mut Parser) -> Result<SpecifiedValue, ()> {
-        use values::specified::{Length, LengthOrPercentage};
-
-        input.try(specified::LengthOrPercentage::parse_non_negative)
-        .or_else(|()| {
-            let ident = try!(input.expect_ident());
-            NoCalcLength::from_str(&ident as &str)
-                         .ok_or(())
-                         .map(specified::LengthOrPercentage::Length)
-        }).map(SpecifiedValue)
+    pub fn parse(_: &ParserContext, input: &mut Parser) -> Result<SpecifiedValue, ()> {
+        use values::specified::{FontRelativeLength, Length, LengthOrPercentage, NoCalcLength};
+        if let Ok(lop) = input.try(specified::LengthOrPercentage::parse_non_negative) {
+            Ok(SpecifiedValue::Length(lop))
+        } else if let Ok(kw) = input.try(KeywordSize::parse) {
+            Ok(SpecifiedValue::Keyword(kw))
+        } else {
+            let ret = match_ignore_ascii_case! {&*input.expect_ident()?,
+                "smaller" => FontRelativeLength::Em(0.85),
+                "larger" => FontRelativeLength::Em(1.2),
```

Comment:
You probably should keep these two as keyword values rather than converting them promptly. CSSOM should get them as keywords given existing browsers' behavior.